### PR TITLE
feat: add network policies reference

### PR DESCRIPTION
**Pull request by SdgJlbl**
Fixes fl-1580

### CODE REVIEW

**Comment by guilhem-barthes** [comment]
Can you move your content to another file and include it in the index (Cf https://github.com/Substra/substra-documentation/pull/424)

File: docs/source/reference/index.rst [hunk 1]
```diff
@@ -19,3 +19,69 @@ All pods can run as non-root, with two exceptions:
...
++--------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------+
+| orchestrator-database    | orchestrator-server                                                                                                                     | NONE                                                                                                                                |
++--------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------+
```
Comment (guilhem-barthes) [inline docs/source/reference/index.rst:37]: Miss `job-migrations (egress orchestrator-database`)

File: docs/source/reference/index.rst [hunk 2]
```diff
@@ -19,3 +19,69 @@ All pods can run as non-root, with two exceptions:
...
+| backend-docker-registry  | backend-task-build, backend-builder, backend-worker, backend-scheduler, backend-scheduler-worker                                        | NONE                                                                                                                                |
++--------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------+
+| backend-server           | backend-worker, backend-builder, backend-server (other orgs)                                                                            | orchestrator-server, backend-server (other orgs), backend-database, backend-redis, backend-object-storage, backend-docker-registry, |
```
Comment (guilhem-barthes) [inline docs/source/reference/index.rst:48]: I don't know if it works saying internet as egress

**Review by guilhem-barthes — COMMENTED** [review]
Thanks for your work! I made some comments on some flows, let me know if you think they are relevant!

File: docs/source/reference/index.rst [hunk 3]
```diff
@@ -19,3 +19,69 @@ All pods can run as non-root, with two exceptions:
...
+| Pod                      | Incoming                                                                                                                                | Outgoing                                                                                                                            |
++==========================+=========================================================================================================================================+=====================================================================================================================================+
+| orchestrator-server      | backend-api-events, backend-worker-events, backend-scheduler, backend-scheduler-worker, backend-server, backend-builder, backend-worker | orchestrator-database                                                                                                               |
```
[the remark this answers is not in the document]
Comment (SdgJlbl) [reply docs/source/reference/index.rst:34]: OK with prefacing that all these connections come from other organisations?
The idea is to give the option, if people deploying the network are using a more advanced network plugin, to know what connections exactly need to be allowed

File: docs/source/reference/index.rst [hunk 4]
```diff
@@ -19,3 +19,69 @@ All pods can run as non-root, with two exceptions:
...
++--------------------------+-----------------------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------+
+| orchestrator-database    | orchestrator-server                                                                                                                     | NONE                                                                                                                                |
```
[the remark this answers is not in the document]
Comment (SdgJlbl) [reply docs/source/reference/index.rst:36]: are there cases where pods in kube are not given access to the DNS resolver?
maybe I can add a footnote to cover all cases?

File: docs/source/reference/index.rst (hunk 2 above)
Comment (SdgJlbl) [reply docs/source/reference/index.rst:48]: how would you phrase it?

**Review by guilhem-barthes — APPROVED** [review]
LGTM

**Pull request merged**